metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,208 @@
+import sys
+
+from metaflow import R
+from metaflow.datastore.datastore import TransformableObject
+from metaflow.decorators import Decorator, StepDecorator
+from metaflow.exception import MetaflowException
+from metaflow.plugins.aws.awslambda.lambda_runner import LAMBDA_MAX_MEMORY_MB
+from metaflow.plugins.aws.batch.batch_decorator import (
+    DEFAULT_TIMEOUT_SECONDS,
+    BatchDecorator,
+    ResourcesDecorator,
+)
+from metaflow.plugins.aws.utils import sync_metadata_to_s3
+from metaflow.plugins.conda.conda_step_decorator import CondaStepDecorator
+from metaflow.plugins.timeout_decorator import get_run_time_limit_for_task
+
+if sys.version_info > (3, 0):
+    from typing import TYPE_CHECKING, Dict, List, Optional
+
+    if TYPE_CHECKING:
+        from metaflow.runtime import CLIArgs
+        from metaflow.graph import FlowGraph
+        from metaflow.package import MetaflowPackage
+        from metaflow.flowspec import FlowSpec
+        from metaflow.datastore import MetaflowDataStore
+        from metaflow.metaflow_environment import MetaflowEnvironment
+        from metaflow.event_logger import EventLogger
+
+from metaflow.metaflow_config import (
+    LAMBDA_IMAGE_URI,
+    LAMBDA_ROLE_ARN,
+)
+
+DEFAULT_TIMEOUT_SECONDS = 600
```

Comment:
This is already available from https://github.com/Netflix/metaflow/pull/463/files#diff-cfe17a874aba5a08a3ed0ae7a9d22caa5eb63caa96bf392481edbdee323fc073R68

Reply:
Why is the default time out set to 10 mins when lambdas run for up to 15 mins?

Reply:
No strong reason. The default timeout on AWS is 3 seconds but that's too short for us (as we need to download the code too). 

On the other hand, if I try to put some really lengthy computation into a Lambda step, it would arguably be better UX if it failed faster and made me think about my choices and Lambda limitations. One could argue that if 10 minutes is not enough for your code, it is likely that 15 wouldn't be enough either.

So 10 minutes is just a point I picked between 3s and 10m. Maybe it should be even shorter?

Reply:
hmm.. If I am executing a flow on Lambda and Metaflow kills the job at 10m interval when I know that Lambdas run till 15 mins - that would be a confusing behavior. The only reason for having a timeout is to prevent long-running expensive jobs (AWS Batch timeout is 5 days for that very reason).